src/edu/washington/escience/myria/api/encoding/QueryConstruct.java

```diff
@@ -85,18 +85,21 @@
       }
     }
 
-    Map<Integer, SubQueryPlan> plan = Maps.newHashMap();
-    Map<PlanFragmentEncoding, RootOperator> instantiatedFragments = Maps.newHashMap();
     Map<Integer, Operator> allOperators = Maps.newHashMap();
     for (PlanFragmentEncoding fragment : fragments) {
-      RootOperator op =
-          instantiateFragment(fragment, args, instantiatedFragments, op2OwnerFragmentMapping, allOperators);
+      instantiateFragmentOperators(fragment, args, allOperators);
+    }
+    while (setConsumerSchema(fragments, allOperators)) {
```

Comment:
Added in 2b76c6b, although it should not happen because once `setConsumerSchema()` returns `false`, we exit.